source/ios/AdaptiveCards/AdaptiveCards/AdaptiveCards/ACRImageProperties.mm

```diff
@@ -98,4 +98,21 @@ - (void)updateContentSize:(CGSize)size
     }
 }
 
++ (ACRAspectRatio)convertToAspectRatio:(CGSize)cgsize
+{
+    if (cgsize.width == 0) {
+        return {.widthToHeight = 1.0f, .heightToWidth = 1.0f};
+    }
+    // keeping all precisions are not necessary, and 744W X 84H and its multiples cause a crash.
+    // rounding off to 100th points.
+    // To make the constraints work, the rounded value has to become 1 when the two ratios are multiplied
+    const CGFloat precision = 100;
+    // MAX is necessary to prevent heightByWidth becoming zero.
+    NSLog(@"%f", round(precision * (cgsize.height / cgsize.width)) / precision);
+    ACRAspectRatio aspectRatio;
+    aspectRatio.heightToWidth = MAX(round(precision * (cgsize.height / cgsize.width)) / precision, 1 / precision);
+    aspectRatio.widthToHeight = 1 / aspectRatio.heightToWidth;
```

Comment:
do widthToHeight need to round too?

Reply:
No, LN114 is there to make  the multiplication of two ratios become one.